File: bootstrap-ui-tests/src/test/java/uitest/pages/RegistrationPage.java
package uitest.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import uitest.commands.SeleniumCommands;

public class RegistrationPage {

    private final WebDriver driver;
    SeleniumCommands sc;

    @FindBy(css = "#login")
    private WebElement loginField;

    @FindBy(css = "#email")
    private WebElement emailField;

    @FindBy(css = "#password")
    private WebElement passwordField;

    @FindBy(css = "#repeatPassword")
    private WebElement repeatPassField;

    @FindBy(css = "button[type=submit]")
    private WebElement registerButton;

    public RegistrationPage(WebDriver driver) {
        this.driver = driver;
        sc = new SeleniumCommands(driver);
    }

    public void register(String login, String email, String password) throws Exception {
        openRegistrationPage();

        loginField.sendKeys(login);
        emailField.sendKeys(email);
        passwordField.sendKeys(password);
        repeatPassField.sendKeys(password);
        registerButton.click();
        sc.waitForFinishLoading();
    }

    public void openRegistrationPage() throws Exception {
        driver.get("http://localhost:8080/#/register");
        sc.waitForFinishLoading();
    }
}

File: bootstrap-ui-tests/src/test/scala/uitest/pages/RegistrationPage.scala
package uitest.pages

import org.openqa.selenium.WebDriver
import org.openqa.selenium.WebElement
import org.openqa.selenium.support.FindBy
import uitest.commands.SeleniumCommands

class RegistrationPage(driver: WebDriver) {
  private val sc: SeleniumCommands = new SeleniumCommands(driver)
  val url = sc.URL + "register"

  @FindBy(css = "#login")
  val loginField: WebElement = null

  @FindBy(css = "#email")
  val emailField: WebElement = null

  @FindBy(css = "#password")
  val passwordField: WebElement = null

  @FindBy(css = "#repeatPassword")
  val repeatPassField: WebElement = null

  @FindBy(css = "button[type=submit]")
  val registerButton: WebElement = null

  def register(login: String, email: String, password: String) {
    openRegistrationPage()
    loginField.sendKeys(login)
    emailField.sendKeys(email)
    passwordField.sendKeys(password)
    repeatPassField.sendKeys(password)
    registerButton.click()
    sc.waitForFinishLoading()
  }

  def openRegistrationPage() {
    driver.get(url)
    sc.waitForFinishLoading()
  }
}
